### ISSUE TITLE
Review client error handling

### ISSUE THREAD

**Issue by febbraiod** [exhi/hoist-core#21]
Client error msgs are/were sometimes being saved as the string "null" but should not. Also have seen the server returning messages as  {msg: "[Ljava.lang.String;@e97529"} .

**Comment by febbraiod** [comment]
The 'null' string message being recorded here looks like it's coming from PIQ-React or Hoist-React. I don't see in PIQ-prod, and the records I'm seeing locally with that particular problem have 1.0-SNAPSHOT as the version. (As recently as this morning)

Will look into that then revisit the {msg: "[Ljava.lang.String;@e97529"} problem later.

**Comment by febbraiod** [comment]
As I suspected, Hoist-React is responsible for this 'bug'. Ultimately, our new fetch api and the old Ext.Ajax.request are not handling the params in the same way. When the {msg: null} gets to the server it is an empty string in one case, and the string 'null' in another.

The was some debate in the MMK office as to whether to look for this string and convert to null on the server before the ClientError is saved or to adjust the ErrorTrackingService on the frontend to send the empty string in lieu of null in the first place.

Both solutions seemed to have merit. The server's should be able to deal with a client app using fetch to send null. The frontend shouldn't create the problem in the first place. I pushed a branch for each.
